test/frontend-integration-test/helloworld.spec.js

```diff
@@ -135,31 +140,35 @@ describe('deploy helloworld sample run', () => {
     assert(attempts, 'waited for 30 seconds but run did not start.');
 
     assert.equal($$('.tableRow').length, 1, 'should only show one run');
-
-    // Navigate to details of the deployed run by clicking its anchor element
-    browser.execute('document.querySelector(".tableRow a").click()');
-  });
-
-  it('switches to config tab', () => {
-    $('button=Config').waitForVisible(waitTimeout);
-    $('button=Config').click();
   });
 
+  // Wait for a reasonable amount of time until the run is done
+  // and navigate to the run details page
   it('waits for run to finish', () => {
-    let status = getValueFromDetailsTable('Status');
-
     let attempts = 0;
-    const maxAttempts = 60;
+    const maxAttempts = 90;
 
-    // Wait for a reasonable amount of time until the run is done
-    while (attempts < maxAttempts && status.trim() !== 'Succeeded') {
+    while (attempts < maxAttempts) {
       browser.pause(1000);
-      status = getValueFromDetailsTable('Status');
       attempts++;
     }
 
-    assert(attempts < maxAttempts, `waited for ${maxAttempts} seconds but run did not succeed. ` +
-      'Current status is: ' + status);
+    browser.execute('document.querySelector(".tableRow a").click()');
+    browser.waitUntil(() => {
+      return new URL(browser.getUrl()).hash.startsWith('#/runs/details');
+    }, waitTimeout);
```

Comment:
Waiting for the run details page to load is technically correct but doesn't change anything in the current setup, because `$('button=Config').waitForVisible();` below would also ensure the page to load--unless there's no config button on run list page.

Reply:
To add my 5 cents: waiting for the page is more explicit. I reverted the change as you suggested above.

Reply:
Absolutely agree explicit is better, and that's what I meant by "technically correct". For this PR, I think it's better to focus on what it claims, the effective change in the `helloworld.spec.js` can be reduced into one-line change of waiting for 90 seconds before clicking into "config" tab, and I was a bit skeptical whether that would fix the test failure we saw. For that reason I suggested we don't make this not-so-relevant change. 
If we were to change this file in a meaningful way again, say in your large PR, I would support waiting explicitly on the page load.